packages/scripts/src/upgrade-tests/voting2/0_Deploy.ts

```diff
@@ -62,21 +62,28 @@ async function main() {
 
   console.log("2. DEPLOYING VOTING V2");
   const votingV2Factory: VotingV2Ethers__factory = await getContractFactory("VotingV2");
-  const emissionRate = "640000000000000000"; // 0.64 UMA per second.
+  const emissionRate = "0"; // Initially set the emission rate to 0.
   const spamDeletionProposalBond = hre.web3.utils.toWei("10000", "ether");
   const unstakeCooldown = 60 * 60 * 24 * 7; // 7 days
   const phaseLength = "86400";
   const minRollToNextRoundLength = "7200";
   const gat = hre.web3.utils.toBN(hre.web3.utils.toWei("5500000", "ether")); // Set the GAT to 5.5 million tokens.
 
+  const requestPriceEvents = await existingVoting.queryFilter(
+    existingVoting.filters.PriceRequestAdded(null, null, null)
+  );
+  const startingIndex = requestPriceEvents.length;
+
+  console.log("Starting index for new Voting contract: ", startingIndex);
+
   const votingV2 = await votingV2Factory.deploy(
     emissionRate,
     spamDeletionProposalBond,
     unstakeCooldown,
     phaseLength,
     minRollToNextRoundLength,
     gat.toString(),
-    "0", // Starting request index of 0 (no offset). TODO change this to the correct number
+    startingIndex, // Starting index for new Voting contract set to the number of existing PriceRequestAdded events.
```

Comment:
Is it ever safe to start with a 0 starting index? It can't roll, right, so that would be considered a bad initial value?

For that reason, should we add a max(startingIndex, 1) or assert(startingIndex > 0) to ensure that we never start with 0?

Reply:
I agree with you @mrice32 that it's logical to add that require - allowing startingIndex=0 is not ideal because if it rolls it breaks the DVM.

However there is some friction if we want to add the assert in the VotingV2 contract:
- The tests use this startingIndex=0 everywhere (and tests never roll the first request). 
- There is very few remaining free bytecode space in the VotingV2 contract, adding that require may not be possible.